src/coreclr/vm/reflectioninvocation.cpp

```diff
@@ -1022,6 +1022,17 @@ FCIMPL5(Object*, RuntimeMethodHandle::InvokeMethod,
         }
 
         InvokeUtil::CopyArg(th, &objs->GetAt(i), &argDest);
+
+#if defined(TARGET_LOONGARCH64)
+        if ((ofs < 0) && (objs->GetAt(i) != 0) && (th.GetVerifierCorElementType() == ELEMENT_TYPE_VALUETYPE) && (th.GetSize() == 8) && (argit.m_idxFPReg == 2))
```

Comment:
What is the situation that this is trying to handle? It does not look right to have this logic here - it seems to be compensating for more fundamental bug somewhere else.

Reply:
This is needed for the type ABI liking LoongArch64.
Here it handled the argment of struct {float a; float b;} which using two float-registers passing the arguments.
We have to treat this struct as two arguments so split it.

Reply:
I understand that these structs require special handling for LoongArch64.

This place does not look like the right place to add this special handling. It should be done by changing the implementation of `ArgDestination` and  `UnBoxIntoArg` instead.

Also, according to https://github.com/loongson/LoongArch-Documentation/pull/32#discussion_r773539198, I think the structures with two doubles need the same treatment.

Reply:
> Also, according to [loongson/LoongArch-Documentation#32 (comment)](https://github.com/loongson/LoongArch-Documentation/pull/32#discussion_r773539198), I think the structures with two doubles need the same treatment.

Double doesn't need because it is a pointer size.

> This place does not look like the right place to add this special handling. It should be done by changing the implementation of `ArgDestination` and `UnBoxIntoArg` instead.

I didn't do that because there are not enought info to do it within the ArgDestination.
For example, ArgDestination can't get the info of the struct.

Reply:
You can modify `ArgDestination` and `ArgLocDesc` as necessary for `LoongArch64` to capture the necessary details. Similar to how these types are modified for Linux x64 ABI.

Reply:
> You can modify `ArgDestination` and `ArgLocDesc` as necessary for `LoongArch64` to capture the necessary details. Similar to how these types are modified for Linux x64 ABI.

I can not modify the `ArgDestination` for LoongArch64 only before you accepting to do it.
Now If I modify the `ArgDestination`, there will be some modification that related with other architecuturs.
Should I do these  within this PR ?

Reply:
I think it would be a good idea to have a separate PR with the VM calling convention related changes: modify `ArgDestination` and `ArgLocDesc`, add code to sign-extend and zero-extended CallDescr worker arguments as necessary, etc.

Reply:
That means I create a new PR for modifing ArgDestination and ArgLocDesc,
this PR should wait for the modify of ArgDestination and ArgLocDesc` finished.

How should I commit new patch based on the modify of ArgDestination and ArgLocDesc` ?
Because this PR is based on a old main, how can I resolve the conflics.

Reply:
That means I create a new PR for modifing ArgDestination and ArgLocDesc,
this PR should wait for the modify of ArgDestination and ArgLocDesc` finished.

How should I commit new patch based on the modify of ArgDestination and ArgLocDesc` ?
Because this PR is based on a old main, how can I resolve the conflics.

Reply:
I had deleted the modification within the file `src/coreclr/vm/reflectioninvocation.cpp`
by the commit
https://github.com/dotnet/runtime/pull/62885/commits/7757e09a9c190813241e7ac02b3436e1e9001471